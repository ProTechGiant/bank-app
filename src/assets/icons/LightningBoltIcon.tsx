import Svg, { Path } from "react-native-svg";

import type { IconProps } from "@/assets/icons";

export function LightningBoltIcon({ width = 12, height = 18, color = "#1E1A25" }: IconProps) {
  return (
    <Svg width={width} height={height} viewBox="0 0 24 25" fill="none" xmlns="http://www.w3.org/2000/svg">
      <Path
        d="M9.19719 22.65C9.04719 22.8 8.86385 22.875 8.64719 22.875C8.43052 22.875 8.23885 22.7917 8.07219 22.625C7.93885 22.4917 7.86385 22.321 7.84719 22.113C7.83052 21.9043 7.88052 21.725 7.99719 21.575L13.0222 14.575L4.64719 13.775C4.19719 13.725 3.89719 13.4873 3.74719 13.062C3.59719 12.6373 3.69719 12.275 4.04719 11.975L14.7972 2.35C14.9472 2.2 15.1305 2.125 15.3472 2.125C15.5639 2.125 15.7555 2.20833 15.9222 2.375C16.0555 2.50833 16.1305 2.679 16.1472 2.887C16.1639 3.09567 16.1139 3.275 15.9972 3.425L10.9722 10.425L19.3472 11.225C19.7972 11.275 20.0972 11.5127 20.2472 11.938C20.3972 12.3627 20.2972 12.725 19.9472 13.025L9.19719 22.65Z"
        fill={color}
      />
    </Svg>
  );
}
