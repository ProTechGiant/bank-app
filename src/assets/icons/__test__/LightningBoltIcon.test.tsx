import { create } from "react-test-renderer";

import { LightningBoltIcon } from "../LightningBoltIcon";

describe("icon/LightningBoltIcon", () => {
  it("renders a LightningBoltIcon", () => {
    const result = create(<LightningBoltIcon />);

    expect(result).toMatchSnapshot();
  });
});
