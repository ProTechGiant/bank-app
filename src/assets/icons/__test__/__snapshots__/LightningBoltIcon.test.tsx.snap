// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon/LightningBoltIcon renders a LightningBoltIcon 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="18"
  bbWidth="12"
  fill="#FF7512"
  focusable={false}
  height={18}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 18,
        "width": 12,
      },
    ]
  }
  vbHeight={18}
  vbWidth={12}
  width={12}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={
      {
        "payload": 4294931730,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M4.99558 18H3.99558L4.99558 11H1.49558C0.615577 11 1.16558 10.25 1.18558 10.22C2.47558 7.94 4.41558 4.54 7.00558 0H8.00558L7.00558 7H10.5156C10.9156 7 11.1356 7.19 10.9156 7.66C6.96558 14.55 4.99558 18 4.99558 18Z"
      fill={
        {
          "payload": 4294931730,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
